core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.html

```diff
@@ -28,26 +35,60 @@ <h4 class="modal-title">Execution results of this workflow</h4>
           ></i>
         </td>
         <td nzEllipsis>{{(i + 1)}}</td>
-        <td nzEllipsis>{{row.startingTime | date:'MM/dd/yyyy HH:mm:ss'}}</td>
-        <td nzEllipsis>{{row.completionTime | date:'MM/dd/yyyy HH:mm:ss'}}</td>
         <td nzEllipsis>
-          <span
-            [ngbTooltip]="getExecutionStatus(row.status)[0]"
-            placement="auto"
-            container="body"
-            class="icon-container"
+          <label *ngIf="workflowExecutionsIsEditingName.indexOf(i) === -1; else customeWorkflow " class="workflow-name"
+            >{{ row.name }}</label
+          >
+          <ng-template #customeWorkflow>
+            <input
+              #customName
+              (focusout)="confirmUpdateWorkflowExecutionsCustomName(row, customName.value, i)"
+              (keyup.enter)="confirmUpdateWorkflowExecutionsCustomName(row, customName.value, i)"
+              placeholder="{{ row.name }}"
+              value="{{ row.name }}"
+            />
+          </ng-template>
+          <button
+            (click)="workflowExecutionsIsEditingName.push(i)"
+            nz-button
+            nzSize="small"
+            nzType="text"
+            class="rename-icon"
           >
             <i
               nz-icon
               nzTheme="outline"
-              [nzType]="getExecutionStatus(row.status)[1]"
-              [ngStyle]="{'color': getExecutionStatus(row.status)[2]}"
-              class="status-icon"
+              ngbTooltip="Custom Execution#{{i}} Name"
```

Comment:
checkout this nz-tooltip usage https://github.com/Texera/texera/blob/0f63d6599121f3302e54e5258309596a1d121362/core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.html#L286
Also, change the label to something like "rename execution {{row.name}}"

Reply:
nz-tooltip will be solved later. Label has been changed